### DISCUSSION TITLE
How to find a div id in a HTML email.body ?

### DISCUSSION THREAD

**Discussion by rudyhadoux** [cypress-io/cypress#15938] [Questions and Help]
Hi,

How to find a div id in a HTML email.body ?

**Comment by bahmutov** [comment]
Not sure what you are asking, can you give more details, please?

- https://on.cypress.io/get
- https://glebbahmutov.com/cypress-examples/7.0.1/commands/querying.html

**Reply by rudyhadoux** [reply]
I used Mailslurp but I prefer free gmail-tester. Unfortunately gmail-tester is hard to configure. With Mailslurp I do not manage to get email.body in html format to use cy.get(#ID).

**Reply by bahmutov** [reply]
Unfortunately, I still do not understand the problem since I have not used Mailslurp / gmail-tester. Maybe the question should be addressed to those projects?

**Comment by rudyhadoux** [comment]
I think it is a Cypress question.

**Reply by bahmutov** [reply]
Can you then create a reproducible repo? Showing a Cypress test failing to do what you expect it to do?

**Comment by rudyhadoux** [comment]
It is a generic question. With any provider, how to transform email.body in a HTML format and how to get div id content ?

**Reply by bahmutov** [reply]
so you get an HTML email and then want to "import" it into Cypress and verify certain elements?

**Reply by rudyhadoux** [reply]
Yes. I have email.body that is a big String.
Maybe we must put it in a temporary HTML file and cy.visit after ?

**Reply by bahmutov** [reply]
You could do that and then use `cy.visit('email.html')`. You could also use `cy.document().invoke('write', emailString)` to place the HTML into the document and then you could interact with it as if it were visited

**Reply by rudyhadoux** [reply]
Thanks. That is the good answer.

**Reply by rudyhadoux** [reply]
It seems to work too :
const body = emails[0].body.html;
